### ISSUE TITLE
NoNameShadowing false positive

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#3787]
## Expected Behavior
```kotlin
list?.let {
  it.map { Unit }
}
```

Should not be flagged. We are shadowing the `it` from `map` but we aren't using it so there is not potential bug here.

## Observed Behavior
That code is flagged

## Your Environment
* Version of detekt used: 1.17.0
